### ISSUE TITLE
TensorFlow Serving fails to load local model

### ISSUE THREAD

**Issue by Camuslu** [tensorflow/serving#697]
I'm new to TensorFlow Serving and trying to load a saved model in my local directory, using `tensorflow_model_server`

I followed the example at the official documentation:
https://www.tensorflow.org/serving/serving_basic#load_exported_model_with_standard_tensorflow_modelserver

After I train the mnist model, it's exported and saved in `/tmp/mnist_model`  

I confirmed that the model is there:

`ls /tmp/mnist_model/1`

returns

> saved_model.pb	variables

Now I try to load the model, again following the documentation:

`tensorflow_model_server --port=9000 --model_name=mnist --model_base_path=/tmp/mnist_model/`

and I get the error 

> FileSystemStoragePathSource encountered a file-system access error:
> Could not find base path /tmp/mnist_model/ for servable mnist

In other posts I've seen people encounter this error when loading model saved on S3 or other remote places. However here the model is on my local directory and I cannot figure out what's causing the error. Has anyone encountered this?

**Comment by chrisolston** [comment]
Hmmm, that's an odd one. Be sure the file permissions are set up right. Also, maybe try without the trailing slash (unlikely to matter, but worth trying).

The code that's triggering this error is pretty straightforward:

if (!Env::Default()->FileExists(servable.base_path()).ok()) {
    return errors::InvalidArgument("Could not find base path ",
                                   servable.base_path(), " for servable ",
                                   servable.servable_name());
  }

**Comment by Camuslu** [comment]
Thanks @chrisolston  
I tried without the trailing slash and it gives the same error.

To add some details:

I started TensorFlow Serving via Docker. I used the `Dockerfile.devel` from https://www.tensorflow.org/serving/docker 

and followed the steps listed there. 


To confirm that I do have the models saved locally:

![screen shot 2017-12-14 at 3 14 13 pm](https://user-images.githubusercontent.com/8095464/34012487-ebf607ce-e0e1-11e7-8314-e613c34b315d.png)

And the error I get:

![screen shot 2017-12-14 at 3 43 49 pm](https://user-images.githubusercontent.com/8095464/34013492-b5716834-e0e5-11e7-9709-3b366a1eff59.png)

(In the screenshot above, the `/apps/tensorflow`is a WORKDIR I specified in the Dockerfile)

**Comment by chrisolston** [comment]
To clarify what I meant: be sure the unix file permissions are set such that the tensorflow-serving server has permission to read those files. You can use "ls -l" to see the current permissions, and "chmod" to change them if needed.

**Comment by Camuslu** [comment]
Thanks! @chrisolston 

After digging around, I added `chmod 777` to those saved model files. That alone didn't solve the issue.

However, what seems to really cause the issue is that, in my local machine the models are saved under `/tmp`, but when I use `docker run` to launch the image (so as to use tensorflow-serving), the root directory is a different one and `/tmp` **no longer has the models**
![screen shot 2017-12-14 at 5 56 19 pm](https://user-images.githubusercontent.com/8095464/34018506-21637d04-e0f8-11e7-9ef9-50a7207cf1b1.png)

So I tried first copying the models to my local directory (before I do `docker run ..`), and they would still be here after I launch docker. 
Then I changed the `base_path` to "/apps/tensorflow/saved_models" where "/apps/tensorflow/" is the WORKDIR I specified in my Dockerfile.

i.e. when I do
`tensorflow_model_server --port=9000 --model_base_path=/apps/tensorflow/saved_models`

It works fine (finally!)

![screen shot 2017-12-14 at 6 00 31 pm](https://user-images.githubusercontent.com/8095464/34018639-c733a56a-e0f8-11e7-8832-178ce087576b.png)

I'm new to docker and the concepts such as image and containers. Again thanks for the help along the thread!
